### ISSUE TITLE
JS error when tapping "Add to Firefox" button for a search tool on Android

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/olympia#1454]
Prerequisites:
Set remote debugging for Android (https://developer.mozilla.org/en-US/docs/Tools/Remote_Debugging/Debugging_Firefox_for_Android_with_WebIDE )

Steps to reproduce:
1. Load AMO-stage on your android device
2. Tap the "Add to Firefox" installation button for a search tool (i.e. RiandkCGI)
3. Observe the messages from the console

Expected results:
There are no issues.

Actual results:
"TypeError: r is not a function" message appears in the console.

Notes/Issues:
Verified on FF43(Android 5.0.1). Issue is reproducing on AMO-stage, dev and prod.
Screenshot for this issue: 
![2016-01-20_1636](https://cloud.githubusercontent.com/assets/15685960/12452668/562f03e2-bf98-11e5-98dc-00e567d77491.png)

**Comment by muffinresearch** [comment]
There's this bit of code here [1]
```js
if (attr.search) {
    install = z.installSearch;
} else {
   install = z.installAddon;
}
install(attr.name, href, attr.icon, hash);
```

Note that install is set to z.installSearch and when install is called it's passed 4 args.

But the installSearch func in zamboni/mobile/buttons.js looks like this:

```js
    z.installSearch = function(name, url, icon, hash, callback) {
        if (window.external && window.external.AddSearchProvider) {
            window.external.AddSearchProvider(url);
            callback();
            _gaq.push(['_trackEvent', 'AMO Addon / Theme Installs', 'addon', name]);
        } else {
            // Alert!  Deal with it.
            alert(gettext('Sorry, you need a Mozilla-based browser (such as Firefox) to install a search plugin.'));
        }
    };
```

And expects 5 args. Hence why the unminified code shows an error: 'callback is not a function'. 

A quick fix will be to make the callback conditional if it's defined.

[1]  https://github.com/mozilla/olympia/blob/fc3728b23d73a23004ab342934af861779094e60/static/js/zamboni/mobile/buttons.js#L106

**Comment by ValentinaPC** [comment]
Verified as fixed on FF43 (Android 5.0.1) on AMO-dev.
Postfix screenshot: 
![2016-01-21_1123](https://cloud.githubusercontent.com/assets/15685960/12476379/97c57cd4-c031-11e5-834a-163e2f5e866a.png)
